### PR TITLE
Add WebView2

### PR DESCRIPTION
**Pull request by bgavrilMS**
- [x] WebView2 logic on .NET 5
- [x] WebUi Factory logic 
- [x] Add support to NET Classic and Net Core 3.1 via Desktop package

# Functionality

## MSAL standalone

|  	|  Embedded WebView	| Default WebView 	|  	
|-	|-	|-	|
|  NET Fx (no change)	        | Legacy 	|  Embedded 		|  
|  NET Core (no change)	| N/A 	| System  	|  	
|  NET5-WIn	|  WebView2  only |  Embedded 	|  	

## MSAL + MSAL.Desktop

Enabled by adding `pcaBuilder.WithDesktopFeatures()`, which also enabled WAM on these languages.

|  	|  Embedded WebView	| Default WebView 	|  	
|-	|-	|-	|
|  NET Fx         | WebView2, fallback to Legacy |  Embedded 		|  
|  NET Core 	| WebView2 only	| Embedded |  	

@henrik-me @jmprieur - could you please validate this behaviour in the tables above let me know if it's ok ?

### Testing: 

- Platforms: net5, netfx, net core
- App type: console, wpf, win forms
- Other logic: launch from MTA thread, from STA thread, with / without capturing SyncronizationContext (console will not have it)

Review help: please test commit by commit. You can skip refacotring commints.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -155,11 +156,13 @@
     <Compile Include="$(PathToMsalSources)\Platforms\Features\DefaultOSBrowser\**\*.cs" />
     <Compile Include="$(PathToMsalSources)\Platforms\Features\WamBroker\**\*.cs" />
     <Compile Include="$(PathToMsalSources)\Platforms\Features\WinFormsWebUi\**\*.cs" />
+    <Compile Include="$(PathToMsalSources)\Platforms\Features\WebView2WebUi\**\*.cs" />
     <Compile Include="$(PathToMsalSources)\Platforms\Features\win32\**\*.cs" />
 
     <!-- This package reference is here to fix a bug in the WinRT interop. 
     It should not be required after the December release of .NET 5 SDK -->
     <PackageReference Include="Microsoft.Windows.CsWinRT" Version="1.1.0" />
+    <PackageReference Include="Microsoft.Web.WebView2" Version="1.0.705.50" />
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:165]: Another COM package, we will probably not be able to add it to NET Classic

File: src/client/Microsoft.Identity.Client/MsalError.cs [hunk 2]
```diff
@@ -943,5 +943,12 @@ public static class MsalError
         /// pops up
         /// </summary>
         public const string WamPickerError = "wam_interactive_picker_error";
+
+        /// <summary>
+        /// <para>What happens?</para>The embedded browser cannot be started because a runtime component is missing.
+        /// <para>Mitigation</para>"The embedded browser needs WebView2 runtime to be installed. An end user of the app can download and install the WebView2 runtime from https://go.microsoft.com/fwlink/p/?LinkId=2124703 and restart the app.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/MsalError.cs:949]:
```suggestion
        /// <para>Mitigation</para>The embedded browser needs WebView2 runtime to be installed. An end user of the app can download and install the WebView2 runtime from https://go.microsoft.com/fwlink/p/?LinkId=2124703 and restart the app.
```

**Comment by jennyf19** [comment]
side note...we don't need this anymore, as they (vs) will use WAM right? [https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/907](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/907)

**Comment by jennyf19** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

does this need to be repeated in both? or linked?

---
Refers to: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsLegacyWebUi/Win32Window.cs:1 in 7885b59. [](commit_id = 7885b5930b3013c8c8e2fe87d3593d4569975a3a, deletion_comment = False)

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Pull request merged**